### PR TITLE
Move dns propagnation helpers to own packages

### PR DESCRIPTION
**Pull request by prksu**
**What this PR does / why we need it**:
This PR move DNS propagation helpers to own packages since `cloud` directory is devoted to related cloud provider stuff

**Special notes for your reviewer**:
/cc @gottwald 
/assign @timoreimann 

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by prksu** [comment]
cc @gottwald

**Review by gottwald — COMMENTED** [review]
Mostly LGTM, however the validation pattern of the "Name" should be extended.

File: api/v1alpha3/docluster_types.go [hunk 1]
```diff
@@ -62,10 +62,13 @@ type DOClusterStatus struct {
...
 type DOControlPlaneDNS struct {
-	// Domain is the DO domain that this record should live in.
-	// It must be pre-existing in your DO account.
+	// Domain is the DO domain that this record should live in. It must be pre-existing in your DO account.
+	// The format must be a string that conforms to the definition of a subdomain in DNS (RFC 1123)
+	// +kubebuilder:validation:Pattern:=^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
 	Domain string `json:"domain"`
 	// Name is the DNS short name of the record (non-FQDN)
+	// The format must be a string that conforms to the definition of a label in DNS (RFC 1123).
+	// +kubebuilder:validation:Pattern:=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
```
[the remark this answers is not in the document]
Comment (prksu) [reply api/v1alpha3/docluster_types.go]: +1

**Review by gottwald — APPROVED** [review]
LGTM

**Review by cpanato — APPROVED** [review]
/lgtm

**Pull request merged**